packages/zipkin/src/request.js

```diff
@@ -16,8 +17,9 @@ function appendZipkinHeaders(req, traceId) {
 }
 
 function addZipkinHeaders(req, traceId) {
-  const headers = appendZipkinHeaders(req, traceId);
-  return Object.assign({}, req, {headers});
+  req.headers = appendZipkinHeaders(req, traceId);
+
+  return req;
```

Comment:
Request is supposed to be immutable, no middleware is supposed to change a request. Could you please explain the intention behind this change?